src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexExclusive(index);
 
+        grow();
+
+        if (index == size) {
+            values[size++] = value;
+        } else {
+            System.arraycopy(values, index, values, index + 1, size - index);
+            values[index] = value;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            T element = list.get(i);
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexInclusive(index);
+
+        return (T) values[index];
     }
 
     @Override
     public void set(T value, int index) {
+        checkIndexInclusive(index);
 
+        values[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexInclusive(index);
+
+        final T removedValue = (T)values[index];
+
+        for (int i = index; i < size - 1; i++) {
+            values[i] = values[i + 1];
+        }
```

Comment:
Use System.arraycopy

Reply:
not fixed
![image](https://github.com/mate-academy/jv-array-list/assets/73041091/637e474e-5b66-4e99-ab3c-684b7bd2f250)